superset/reports/commands/execute.py

```diff
@@ -330,13 +332,25 @@ def _get_log_data(self) -> HeaderDataType:
         }
         return log_data
 
+    def _get_xlsx_data(self) -> bytes:
+        csv_data = self._get_csv_data()
+
+        df = pd.read_csv(BytesIO(csv_data))
```

Comment:
Retrieving `df` from csv data it's not the good approach. We need to write a common method for csv and excel where we provide `df` param like `get_data(self, df: pd.DataFrame)` in `superset/common/query_context_processor.py`.
@dpgaspar Am I correct?
![image](https://user-images.githubusercontent.com/29536522/215264995-1cfd07c4-1c12-43b8-8d03-2920b40d0ce1.png)

Reply:
I am a bit confused about this part. We are getting df in order to convert it to excel. 

I mean if we already have df we can just convert to excel so what would the common method achieve? Could you provide more details?